### PR TITLE
[CONTP-472] Configure gpu tagging to search for names using prefixes

### PR DESCRIPTION
**Pull request by gabedos**
### What does this PR do?

Agent GPU Tagging to search for GPU resources using prefixes.

### Motivation

Support different types of kubernetes resource request definitions. For example, nvidia.com/gpu and nvidia.com/mig-1g.10gb

### Describe how to test/QA your changes

Deploy the agent with a default configuration such as:
```
datadog:
  kubelet:
    tlsVerify: false
  clusterName: "INSERT_CLUSTER_NAME"
agents:
  image:
    repository: INSERT_REPO_NAME
    tag: INSERT_TAG_NAME
    pullPolicy: Always
```
Deploy a dummy workload requesting GPU resources. Note, the values must be zero for kubernetes to schedule the pods since you probably don't have GPU resources on your cluster.
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: dummy-nginx-app
spec:
  replicas: 1
  selector:
    matchLabels:
      app: dummy-nginx-app
  template:
    metadata:
      labels:
        app: dummy-nginx-app
    spec:
      containers:
        - name: dummy-nginx-app
          image: nginx
          resources:
            requests:
              memory: "64Mi"
              cpu: "250m"
              nvidia.com/mig-something: "0"
              nvidia.com/gpu: "0"
              amd.com/gpu: "0"
              gpu.intel.com/xe: "0"
            limits:
              memory: "128Mi"
              cpu: "500m"
              nvidia.com/mig-something: "0"
              nvidia.com/gpu: "0"
              amd.com/gpu: "0"
              gpu.intel.com/xe: "0"
```

Then execute the following command and search for the gpu tags on the pod and container

```
k exec -it datadog-agent-linux-h6vkz -n datadog-agent-helm -- agent tagger-list
=== Entity container_id://173f2c6b8102d6ebf76bbbc6ee1ad8126d70f24950e0cff0c77f5d6b6d192414 ===
== Source workloadmeta-container =
=Tags: [container_id:173f2c6b8102d6ebf76bbbc6ee1ad8126d70f24950e0cff0c77f5d6b6d192414 container_name:dummy-nginx-app docker_image:nginx:latest gpu_vendor:amd gpu_vendor:intel gpu_vendor:nvidia image_id:nginx@sha256:28402db69fec7c17e179ea87882667f1e054391138f77ffaf0c3eb388efc3ffb image_name:nginx image_tag:latest short_image:nginx]
== Source workloadmeta-kubernetes_pod =
=Tags: [container_id:173f2c6b8102d6ebf76bbbc6ee1ad8126d70f24950e0cff0c77f5d6b6d192414 display_container_name:dummy-nginx-app_dummy-nginx-app-6d47fbbbc5-mvmqk gpu_vendor:amd gpu_vendor:intel gpu_vendor:nvidia image_id:nginx@sha256:28402db69fec7c17e179ea87882667f1e054391138f77ffaf0c3eb388efc3ffb image_name:nginx image_tag:latest kube_container_name:dummy-nginx-app kube_deployment:dummy-nginx-app kube_namespace:default kube_ownerref_kind:replicaset kube_ownerref_name:dummy-nginx-app-6d47fbbbc5 kube_qos:Burstable kube_replica_set:dummy-nginx-app-6d47fbbbc5 pod_name:dummy-nginx-app-6d47fbbbc5-mvmqk pod_phase:running short_image:nginx]
===
```

### Possible Drawbacks / Trade-offs

N/A

### Additional Notes

N/A

### CODE REVIEW

File: comp/core/workloadmeta/collectors/internal/kubelet/kubelet.go [hunk 1]
```diff
@@ -446,10 +445,20 @@ func extractResources(spec *kubelet.ContainerSpec) workloadmeta.ContainerResourc
...
+	for resourceName := range spec.Resources.Requests {
+		resourceKeys = append(resourceKeys, resourceName)
+	}
+
+	for _, gpuResourcePrefix := range kubelet.GetGPUResourcePrefixes() {
+		for _, resourceKey := range resourceKeys {
+			if strings.HasPrefix(string(resourceKey), string(gpuResourcePrefix)) {
+				if gpuReq, found := spec.Resources.Requests[resourceKey]; found {
+					resources.GPURequest = pointer.Ptr(uint64(gpuReq.Value()))
+					uniqueGPUVendor[extractGPUVendor(gpuResourcePrefix)] = true
```
Comment (clamoriniere) [inline comp/core/workloadmeta/collectors/internal/kubelet/kubelet.go:459]: quick optimization, you could break one of the for loop when you add an item in the `uniqueGPUVendor` map.

**Review by clamoriniere — APPROVED** [review]
looks good to me. 
I'm just wondering if we can find any other type of device starting with the same prefix. 
for example a possible  `nvidia.com/network` device. But I wasn't able to find any real example.

**Comment by gabedos** [comment]
> looks good to me. I'm just wondering if we can find any other type of device starting with the same prefix. for example a possible `nvidia.com/network` device. But I wasn't able to find any real example.

It seems like there actually is a few other nvidia.com/* resources that can be defined so I will probably restrict it to just gpu and mig* https://docs.nvidia.com/networking/display/kubernetes2410/getting+started+with+kubernetes

**Review by mfpierre — APPROVED** [review]

**Pull request merged**